massgen/tests/test_checklist_tools_server.py

```
@@ -561,52 +561,96 @@ def test_report_path_optional(self):
+        assert _checklist_required_true(100, num_items=3) == 2
 
-    def test_threshold_0_requires_all_5_items(self):
-        """At threshold 0 with 5 items, all items should be required."""
+    def test_required_is_monotonic_non_increasing(self):
+        """Higher effective threshold must never increase the required count."""
         from massgen.system_prompt_sections import _checklist_required_true
 
-        assert _checklist_required_true(0, num_items=5) == 5
+        for n in (3, 4, 5):
+            seq = [_checklist_required_true(et, num_items=n) for et in range(0, 11)]
+            assert all(b <= a for a, b in zip(seq, seq[1:])), (n, seq)
```

Ruff (0.15.13)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)